src/btree/col_srch.c

```diff
@@ -120,7 +120,17 @@ restart_page:	page = current->page;
 	page = current->page;
 	cbt->ref = current;
 	cbt->recno = recno;
-	cbt->compare = 0;
+
+	/* 
+	 * Don't bother searching if the caller is appending a new record where
+	 * we'll allocate the record number; we're not going to find a match by
+	 * definition, and we figure out the record number and position when we
+	 * do the work.
+	 */
+	if (recno == UINT64_MAX) {
+		cbt->compare = -1;
```

Comment:
The code below used to setup cbt->slot and cbt->recno in this case. I'm not confident that isn't necessary - can you reassure me?